### PR TITLE
Add SMSC (СМСЦентр) provider notification

### LINKED ISSUE #3334: Add SMSC (СМСЦентр) provider notification
### ⚠️ Please verify that this feature request has NOT been suggested before.

- [X] I checked and didn't find similar feature request

### 🏷️ Feature Request Type

New Notification

### 🔖 Feature description

Add new notification service https://smsc.kz/.
I took the ClickSendSMS code as a base and added additional parameters.
This service allows you to send not only SMS, but I needed to send SMS

### ✔️ Solution

File src/components/notifications/index.js 
after the line `import ClickSendSMS from "./ClickSendSMS.vue";` added `import SMSC from "./SMSC.vue";`
after the line `clicksendsms": ClickSendSMS,` added `"smsc": SMSC,`

File server/notification.js
after the line `const ClickSendSMS = require("./notification-providers/clicksendsms");` added `const SMSC = require("./notification-providers/smsc");`
after the line `new ClickSendSMS(),` added `new SMSC(),`

File src/components/NotificationDialog.vue
after the line `"clicksendsms": "ClickSend SMS",` added `"smsc": "SMSC",`

And new created files server/notification-providers/smsc.js and src/components/notifications/SMSC.vue[smsc.zip](https://github.com/louislam/uptime-kuma/files/11907037/smsc.zip)

Full files in provider
[fullfiles.zip](https://github.com/louislam/uptime-kuma/files/11907084/fullfiles.zip)

### ❓ Alternatives

_No response_

### 📝 Additional Context

With git fork I did not understand the instructions, so I posted them as files

image

### PR DESCRIPTION
**Pull request by louislam**
By @FlatronBuda, closes #3334.

### CODE REVIEW

File: server/notification-providers/smsc.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+    name = "smsc";
+
+    async send(notification, msg, monitorJSON = null, heartbeatJSON = null) {
+        let okMsg = "Sent Successfully.";
+        try {
+            let config = {
+                headers: {
+                    "Content-Type": "application/json",
+                    // "Authorization": "Basic " + Buffer.from(notification.smscLogin + ":" + notification.smscPassword).toString("base64"),
```
Comment (CommanderStorm) [inline server/notification-providers/smsc.js:14]: I don't see the additional value of this line
```suggestion
```

File: server/notification-providers/smsc.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+                "phones=" + notification.smscToNumber,
+                "mes=" + encodeURIComponent(msg.replace(/[^\x00-\x7F]/g, "")),
+            ];
+            if (notification.smscSenderName !== "") {
+                getArray.push("sender=" + notification.smscSenderName);
+            }
+
+            let resp = await axios.get("https://smsc.kz/sys/send.php?" + getArray.join("&"), config);
+            if (resp.data.id === undefined) {
+                let error = "Something gone wrong. Api returned " + resp.data.error + ". Code " + resp.data.error_code;
```
Comment (CommanderStorm) [inline server/notification-providers/smsc.js:33]: I think this could be more readable
```suggestion
                let error = `Something gone wrong. Api returned code ${resp.data.error_code}: ${resp.data.error}`;
```

File: server/notification-providers/smsc.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
+const NotificationProvider = require("./notification-provider");
+const axios = require("axios");
+
+class SMSC extends NotificationProvider {
+
```
Comment (CommanderStorm) [inline server/notification-providers/smsc.js:5]:
```suggestion
```

File: src/components/notifications/SMSC.vue [hunk 4]
```diff
@@ -0,0 +1,43 @@
+<template>
+    <div class="mb-3">
+        <label for="smsc-login" class="form-label">{{ $t("API Username") }}</label>
+        <i18n-t tag="div" class="form-text" keypath="wayToGetClickSendSMSToken">
```
Comment (CommanderStorm) [inline src/components/notifications/SMSC.vue:4]: All the new translations need to be added to `en.json`. Otherwise this is not going to be translatable into other languages.

File: src/components/notifications/SMSC.vue [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+            <a href="https://smsc.kz/" target="_blank">{{ $t("here") }}</a>
+        </i18n-t>
+        <input id="smsc-login" v-model="$parent.notification.smscLogin" type="text" class="form-control" required>
+        <label for="smsc-key" class="form-label">{{ $t("API Key") }}</label>
+        <HiddenInput id="smsc-key" v-model="$parent.notification.smscPassword" :required="true" autocomplete="new-password"></HiddenInput>
+    </div>
+    <div class="mb-3">
+        <div class="form-text">
+            {{ $t("checkPrice", ['СМСЦ']) }}
```
Comment (CommanderStorm) [inline src/components/notifications/SMSC.vue:13]: What is the purpose of `СМСЦ`? Is there an english translation?

File: src/components/notifications/SMSC.vue [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+            <a href="https://smsc.kz/tariffs/" target="_blank">https://smsc.kz/tariffs/</a>
+        </div>
+    </div>
+    <div class="mb-3">
+        <label for="smsc-to-number" class="form-label">{{ $t("Recipient Number") }}</label>
+        <input id="smsc-to-number" v-model="$parent.notification.smscToNumber" type="text" minlength="8" maxlength="14" class="form-control" required>
```
Comment (CommanderStorm) [inline src/components/notifications/SMSC.vue:19]: Is the min/max lenght restricted like this by the Provider?

```suggestion
        <input id="smsc-to-number" v-model="$parent.notification.smscToNumber" type="text" minlength="4" maxlength="15" class="form-control" required>
```
8 might not be the minimum for phone numbers: https://stackoverflow.com/questions/14894899/what-is-the-minimum-length-of-a-valid-international-phone-number

Also 15 are allowed by https://en.wikipedia.org/wiki/Telephone_numbering_plan

File: src/components/notifications/SMSC.vue [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+    </div>
+    <div class="mb-3">
+        <label for="smsc-sender-name" class="form-label">{{ $t("From Name/Number") }}</label>
+        <input id="smsc-sender-name" v-model="$parent.notification.smscSenderName" type="text" minlength="3" maxlength="11" class="form-control">
```
Comment (CommanderStorm) [inline src/components/notifications/SMSC.vue:23]: Is the min/max lenght restricted like this by the Provider?

Names can be longer and some names are as short as 2 Chars afaik
```suggestion
        <input id="smsc-sender-name" v-model="$parent.notification.smscSenderName" type="text" minlength="2" class="form-control">
```

File: src/components/notifications/SMSC.vue [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+        <div class="form-text">{{ $t("Leave blank to use a shared sender number.") }}</div>
+    </div>
+    <div class="mb-3">
+        <label for="smsc-platform" class="form-label">{{ $t("smscTranslit") }}</label><span style="color: red;"><sup>*</sup></span>
+        <select id="smsc-platform" v-model="$parent.notification.smscTranslit" class="form-select">
+            <option value="0">{{ $t("Default") }}</option>
+            <option value="1">Translit</option>
+            <option value="2">MpaHc/Ium</option>
```
Comment (CommanderStorm) [inline src/components/notifications/SMSC.vue:31]: Are these options Translatable?

File: src/components/notifications/SMSC.vue [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+        <label for="smsc-to-number" class="form-label">{{ $t("Recipient Number") }}</label>
+        <input id="smsc-to-number" v-model="$parent.notification.smscToNumber" type="text" minlength="8" maxlength="14" class="form-control" required>
+    </div>
+    <div class="mb-3">
+        <label for="smsc-sender-name" class="form-label">{{ $t("From Name/Number") }}</label>
+        <input id="smsc-sender-name" v-model="$parent.notification.smscSenderName" type="text" minlength="3" maxlength="11" class="form-control">
+        <div class="form-text">{{ $t("Leave blank to use a shared sender number.") }}</div>
+    </div>
+    <div class="mb-3">
+        <label for="smsc-platform" class="form-label">{{ $t("smscTranslit") }}</label><span style="color: red;"><sup>*</sup></span>
```
Comment (CommanderStorm) [inline src/components/notifications/SMSC.vue:27]: Just for clarification: what is a `smscTranslit`?

**Review by CommanderStorm — CHANGES_REQUESTED** [review]
@FlatronBuda
I have left some inline changes. Could you review them?

## A general request
Since I cannot test this notification provider, could you attach screenshots of
- The notificationprovider sending the `Test` event
- The notificationprovider sending the `Monitor down` event (f.ex. setting up monitoring, then changing the url from `google.com` to `does_not_exist.com`)
- The notificationprovider sending the `Monitor up` event (f.ex. then changing the url back to `google.com`)

File: server/notification-providers/smsc.js (hunk 2 above)
Comment (FlatronBuda) [reply server/notification-providers/smsc.js:33]: Agree

File: src/components/notifications/SMSC.vue (hunk 6 above)
Comment (FlatronBuda) [reply src/components/notifications/SMSC.vue:19]: I checked my number. Its length was its 12 characters.

File: src/components/notifications/SMSC.vue (hunk 7 above)
Comment (FlatronBuda) [reply src/components/notifications/SMSC.vue:23]: [Official documentation](https://smsc.kz/api/http/#menu)
paragraph sender
The sender's name displayed on the recipient's phone. English letters, numbers, spaces, and some characters are allowed. Length – 11 characters or 15 digits. All names are registered in your personal account on [this page](https://smsc.kz/senders/).

File: src/components/notifications/SMSC.vue (hunk 8 above)
Comment (FlatronBuda) [reply src/components/notifications/SMSC.vue:31]: The documentation is in English. Top right flag icon.
[Official documentation](https://smsc.kz/api/http/#menu)
paragraph translit. That's what they're called.
A sign that the message you want to translate in translit.
0 (default) – do not translate into transliteration.
1 – translate to translit as " translate".
2 – translate into transliteration as " mpaHc/Ium".

File: src/components/notifications/SMSC.vue (hunk 9 above)
Comment (FlatronBuda) [reply src/components/notifications/SMSC.vue:27]: I think I can just leave the name "Translit" in English. 
image

If you can change these edits I will be grateful

File: src/components/notifications/SMSC.vue (hunk 8 above)
Comment (CommanderStorm) [reply src/components/notifications/SMSC.vue:31]: What does "translate into transliteration" mean?

File: src/components/notifications/SMSC.vue (hunk 7 above)
Comment (CommanderStorm) [reply src/components/notifications/SMSC.vue:23]: Is `Length – 11 characters or 15 digits` a minimum or maximum?

File: src/components/notifications/SMSC.vue (hunk 8 above)
Comment (CommanderStorm) [reply src/components/notifications/SMSC.vue:31]: This does only partially answer my question:
-  is `MpaHc/Ium` translatable into english?
- What provider is this translated by?

File: src/components/notifications/SMSC.vue (hunk 9 above)
Comment (CommanderStorm) [reply src/components/notifications/SMSC.vue:27]: `translit` is not an English word.
I think `smscTranslateInTranslit` could be more appropriate and less confusing for future maintainers.

File: src/components/notifications/SMSC.vue (hunk 5 above)
Comment (CommanderStorm) [reply src/components/notifications/SMSC.vue:13]: Let me rephrase the question:
I am not convinced that our translators will be able to translate this.
What is the purpose of `СМСЦ` in `checkPrice`?

Could you propose a message which is simpler to translate for our fleet of translators? See https://github.com/louislam/uptime-kuma/blob/master/src/lang/README.md

File: src/components/notifications/SMSC.vue (hunk 9 above)
Comment (louislam) [reply src/components/notifications/SMSC.vue:27]: I think SMSC is unlikely to be used outside its country, so translation may not be very important.

File: src/components/notifications/SMSC.vue (hunk 7 above)
Comment (FlatronBuda) [reply src/components/notifications/SMSC.vue:23]: minimum=1 and maximum=15

File: src/components/notifications/SMSC.vue (hunk 5 above)
Comment (FlatronBuda) [reply src/components/notifications/SMSC.vue:13]: СМСЦ - This is the name of the service in Russian. Their correct name is "SMSЦентр" from the logo

File: src/components/notifications/SMSC.vue (hunk 8 above)
Comment (FlatronBuda) [reply src/components/notifications/SMSC.vue:31]: Option 0 (default)
Send SMS Text "Подписаться на новости"
Option 1 (translit)
In Cyrillic, for example, "Подписаться на новости" converts to Latin "Podpisat'sja na novosti" Send SMS. Example service https://translit.ru/
Option 2 (mpaHc/Ium)
I did not use this option.


This is done to reduce the cost of SMS. Since SMS is more expensive in Cyrillic

File: src/components/notifications/SMSC.vue (hunk 9 above)
Comment (FlatronBuda) [reply src/components/notifications/SMSC.vue:27]: Yes, most likely only for Kazakhstan. For 1 country

**Comment by FlatronBuda** [comment]
> @FlatronBuda I have left some inline changes. Could you review them?
> 
> ## A general request
> Since I cannot test this notification provider, could you attach screenshots of
> 
> * The notificationprovider sending the `Test` event
> * The notificationprovider sending the `Monitor down` event (f.ex. setting up monitoring, then changing the url from `google.com` to `does_not_exist.com`)
> * The notificationprovider sending the `Monitor up` event (f.ex. then changing the url back to `google.com`)

Didn't translate fields. Since I created it based on the ClickSendSMS plugin. Everything has already been translated.
image
image

File: src/components/notifications/SMSC.vue (hunk 6 above)
Comment (FlatronBuda) [reply src/components/notifications/SMSC.vue:19]: +7XXXXXXXXXX or 8XXXXXXXXXX minlength="11" and maxlength unlimited. Since you can enter multiple numbers
